pkg/descheduler/controllers/migration/filter.go

```diff
@@ -140,6 +140,49 @@ func (r *Reconciler) filterMaxMigratingPerNamespace(pod *corev1.Pod) bool {
 	return !exceeded
 }
 
+func (r *Reconciler) filterMaxUnavailableByPUB(pod *corev1.Pod) bool {
+	pub, err := r.pubFinder.GetPubForPod(pod)
+	if err != nil {
+		return false
+	}
+	if pub == nil {
+		return true
+	}
+	pods, expectedReplicas, err := r.pubFinder.GetPodsForPub(pub, false)
+	if err != nil {
+		return false
+	}
+	maxUnavailable := pub.Status.UnavailableAllowed
```

Comment:
This implementation approach has a significant issue in the context of concurrent eviction, as the presence of synchronous delay in PUB Informer can result in inaccurate `UnavailableAllowed` data being obtained here.